### ISSUE TITLE
PLC0415: local imports

### ISSUE THREAD

**Issue by smurfix** [astral-sh/ruff#20638]
### Summary

This pattern is used often enough:

``` python
b64decode = None
...
def some_fn():
    global b64decode
    if b64decode is None:
        from base64 import b64decode  # noqa:PLC0415
```

**Comment by amyreese** [comment]
What is the issue here? Are you wanting this sort of pattern to be allowed, but it's currently flagged by the rule? 

I'm guessing this is an attempt at lazy imports, so moving it to global level is undesired. PLC0415 has support for ignoring specific imports if they are listed in `banned-module-level-imports`: https://docs.astral.sh/ruff/rules/import-outside-top-level/#see-also
